### DISCUSSION TITLE
Pulling an Imgui window out of the main glfw window freezes everything

### DISCUSSION THREAD

**Discussion by dhruvthanki** [ocornut/imgui#6952] [New Users Build/Link/Run issues ONLY!]
Hello! First time user here.
I am using the docking branch of v1.89.9 on ubuntu. When I build the example_glfw_opengl3 demo using the following CMakelists.txt file. I run into some weird issues.
```cmake
cmake_minimum_required(VERSION 3.5)
project(imgui_example_glfw_vulkan C CXX)

if(NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE Debug CACHE STRING "" FORCE)
endif()

set(CMAKE_CXX_STANDARD 11)
set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")

# To install GLFW3: sudo apt install libglfw3-dev
find_package(glfw3 REQUIRED)
find_package(Vulkan REQUIRED)
# find_package(implot CONFIG REQUIRED)
find_package(OpenGL REQUIRED)
# find_package(imguizmo CONFIG REQUIRED)

# To install MuJoCo: pip install mujoco
# To find path to mujoco: pip show mujoco
set(MJC /home/dhruv/.local/lib/python3.10/site-packages/mujoco)
message("Mujoco Path : " ${MJC})
include_directories( ${MJC}/include )

# Dear ImGui
set(IMGUI_DIR /home/dhruv/Documents/Github/imgui/)
set(IMGUI imgui)
add_library(${IMGUI} STATIC 
${IMGUI_DIR}/backends/imgui_impl_glfw.cpp ${IMGUI_DIR}/backends/imgui_impl_vulkan.cpp ${IMGUI_DIR}/backends/imgui_impl_opengl3.cpp 
${IMGUI_DIR}/imgui.cpp ${IMGUI_DIR}/imgui_draw.cpp ${IMGUI_DIR}/imgui_demo.cpp ${IMGUI_DIR}/imgui_tables.cpp ${IMGUI_DIR}/imgui_widgets.cpp
)
target_include_directories(${IMGUI} PUBLIC ${IMGUI_DIR} ${IMGUI_DIR}/backends ..)

set(IMPLOT_DIR /home/name/Documents/Github/ImMuJoCo/external/implot)
set(IMPLOT implot)
add_library(${IMPLOT} STATIC 
${IMPLOT_DIR}/implot.cpp
${IMPLOT_DIR}/implot_demo.cpp
${IMPLOT_DIR}/implot_items.cpp 
)
target_include_directories(${IMPLOT} PUBLIC ${IMPLOT_DIR})
target_link_libraries(${IMPLOT} PUBLIC ${IMGUI})


# App
file(GLOB sources *.cpp)
# add_executable(app ${sources})
add_executable(app main.cpp)
target_link_libraries(app libglfw.so.3 Vulkan::Vulkan ${IMGUI} ${OPENGL_opengl_LIBRARY} ${IMPLOT} ${MJC}/libmujoco.so.3.0.0)
```
The only difference with the code on the repo is the following to add multi-viewport functionality.
```cpp
int main()
{
    IMGUI_CHECKVERSION();
    ImGui::CreateContext();
    ImGuiIO& io = ImGui::GetIO(); (void)io;
    io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard;     // Enable Keyboard Controls
    io.ConfigFlags |= ImGuiConfigFlags_NavEnableGamepad;      // Enable Gamepad Controls
    io.ConfigFlags |= ImGuiConfigFlags_ViewportsEnable;
    .......
    while (!glfwWindowShouldClose(window))
    {
    ImGui::UpdatePlatformWindows();
    ImGui::RenderPlatformWindowsDefault();

    glfwSwapBuffers(window);
    }
 // Cleanup
    ImGui_ImplOpenGL3_Shutdown();
    ImGui_ImplGlfw_Shutdown();
    ImGui::DestroyContext();

    glfwDestroyWindow(window);
    glfwTerminate();

    return 0;
}
```
The weird behaviour I see is that, When I pull the imgui window out of the glfw window everything in glfw window freezes but the imgui panel continues to function. And when I pull the imgui panel back into the glfw window, the imgui window disappears everything is still frozen in the glfw window.
Any help would be appreciated. Thank you!

**Comment by dhruvthanki** [comment]
The vulcan example works perfectly well, I only see the problem with opengl example.

**Comment by PathogenDavid** [comment]
There is a significant number of critical function calls missing in the snippet you provided. You either snipped too much or you need to revisit [`example_glfw_opengl3`](https://github.com/ocornut/imgui/blob/2a6d7b1eaa1851e5b53d8c7a361a8d54f5a9ae8c/examples/example_glfw_opengl3/main.cpp) and add all of the stuff you're missing.

Based on the issue you're describing I suspect you're missing a call to `glfwMakeContextCurrent` to restore the main window's context:

https://github.com/ocornut/imgui/blob/2a6d7b1eaa1851e5b53d8c7a361a8d54f5a9ae8c/examples/example_glfw_opengl3/main.cpp#L191-L200

When using the `docking` branch, make sure you reference the examples from the `docking` branch as they are slightly different.

**Reply by dhruvthanki** [reply]
That,s it! That solved it. Thank you very much for the help!
